x-pack/plugins/security_solution/common/api/detection_engine/prebuilt_rules/model/diff/three_way_diff/three_way_diff_outcome.ts

```diff
@@ -70,3 +118,29 @@ export const determineIfValueCanUpdate = (diffCase: ThreeWayDiffOutcome): boolea
     diffCase === ThreeWayDiffOutcome.CustomizedValueCanUpdate
   );
 };
+
+/**
+ * Returns a boolean if 2 arrays contain same the elements agnostic of order after being deduplicated
+ *
+ * NOTE: array equality is case insensitive
+ */
+const arraysHaveSameElements = <T>(arr1: T[], arr2: T[]) => {
+  const set1 = new Set(
+    arr1.map((value) => (typeof value === 'string' ? value.toLowerCase() : value))
+  );
+  const set2 = new Set(
+    arr2.map((value) => (typeof value === 'string' ? value.toLowerCase() : value))
+  );
```

Comment:
We're calling `arraysHaveSameElements` multiple times for the same arrays, thus creating the same sets multiple times. Let's avoid this and create these sets outside of this function - in `determineOrderAgnosticDiffOutcome`. Then, the function would have a signature `<T>(set1: Set<T>, set2: Set<T>): boolean` and could be renamed to `areSetsEqual`.